File: import_services.php
<?php
/**
 * Import servizi agenzie da CSV PULITO a database
 * USA SOLO CSV generato da clean_notion_csv_DEFINITIVO.py
 * Esegui: php import_services_FINAL.php /root/agenzie_clean.csv
 */

require_once 'config/database.php';

echo "=== IMPORT SERVIZI AGENZIE ===\n\n";

// Path CSV
if (isset($argv[1])) {
    $csv_file = $argv[1];
} else {
    $csv_file = readline("Inserisci il path del CSV agenzie PULITO: ");
}

if (!file_exists($csv_file)) {
    die("❌ File non trovato: $csv_file\n");
}

echo "✅ File trovato\n";
echo "📖 Lettura CSV...\n\n";

$pdo = getDB();

// Leggi CSV
$handle = fopen($csv_file, 'r');
$headers = fgetcsv($handle); // Prima riga = headers

echo "📋 Colonne servizi trovate:\n";
$service_columns = ['cb_suite_status', 'canva_status', 'regold_activation', 'james_status', 'docudrop_activation', 'unique_status'];
foreach ($headers as $idx => $header) {
    if (in_array($header, $service_columns)) {
        echo "  [$idx] $header\n";
    }
}

echo "\n";
$confirm = readline("Vuoi procedere con l'import? (si/no): ");

if (strtolower($confirm) !== 'si') {
    die("❌ Import annullato\n");
}

// Helper functions
function parseDate($value) {
    if (empty($value) || $value === 'Empty') return null;
    
    // Formato DD/MM/YYYY
    if (preg_match('/^(\d{2})\/(\d{2})\/(\d{4})$/', $value, $matches)) {
        return $matches[3] . '-' . $matches[2] . '-' . $matches[1];
    }
    
    // Formato YYYY-MM-DD
    if (preg_match('/^\d{4}-\d{2}-\d{2}$/', $value)) {
        return $value;
    }
    
    return null;
}

function isActive($value) {
    if (empty($value) || $value === 'Empty') return 0;
    $value = strtolower(trim($value));
    return in_array($value, ['attivo', 'active', 'si', 'yes', '1', 'true']) ? 1 : 0;
}

function getColumnValue($row, $headers, $columnName) {
    $index = array_search($columnName, $headers);
    return $index !== false ? trim($row[$index]) : '';
}

// Import
$imported = 0;
$errors = 0;
$row_num = 0;

while (($row = fgetcsv($handle)) !== false) {
    $row_num++;
    
    $code = getColumnValue($row, $headers, 'code');
    
    if (empty($code)) {
        continue;
    }
    
    // Trova agency_id
    $stmt = $pdo->prepare("SELECT id FROM agencies WHERE code = ?");
    $stmt->execute([$code]);
    $agency = $stmt->fetch();
    
    if (!$agency) {
        echo "⚠️  Riga $row_num: Agenzia $code non trovata nel database\n";
        $errors++;
        continue;
    }
    
    $agency_id = $agency['id'];
    
    // CB SUITE
    $cb_suite_status = getColumnValue($row, $headers, 'cb_suite_status');
    if (!empty($cb_suite_status) && $cb_suite_status !== 'Empty') {
        $stmt = $pdo->prepare("
            INSERT INTO agency_services (agency_id, service_name, is_active, activation_date, expiration_date, renewal_required, invoice_reference, notes)
            VALUES (?, 'cb_suite', ?, ?, ?, ?, ?, ?)
            ON DUPLICATE KEY UPDATE 
                is_active = VALUES(is_active),
                activation_date = VALUES(activation_date),
                expiration_date = VALUES(expiration_date),
                renewal_required = VALUES(renewal_required),
                invoice_reference = VALUES(invoice_reference),
                notes = VALUES(notes)
        ");
        $stmt->execute([
            $agency_id,
            isActive($cb_suite_status),
            parseDate(getColumnValue($row, $headers, 'cb_suite_start')),
            parseDate(getColumnValue($row, $headers, 'cb_suite_end')),
            getColumnValue($row, $headers, 'cb_suite_renewal'),
            getColumnValue($row, $headers, 'cb_suite_invoice'),
            getColumnValue($row, $headers, 'cb_suite_notes')
        ]);
        $imported++;
    }
    
    // CANVA
    $canva_status = getColumnValue($row, $headers, 'canva_status');
    if (!empty($canva_status) && $canva_status !== 'Empty') {
        $stmt = $pdo->prepare("
            INSERT INTO agency_services (agency_id, service_name, is_active)
            VALUES (?, 'canva', ?)
            ON DUPLICATE KEY UPDATE is_active = VALUES(is_active)
        ");
        $stmt->execute([$agency_id, isActive($canva_status)]);
        $imported++;
    }
    
    // REGOLD
    $regold_activation = getColumnValue($row, $headers, 'regold_activation');
    if (!empty($regold_activation) && $regold_activation !== 'Empty') {
        $stmt = $pdo->prepare("
            INSERT INTO agency_services (agency_id, service_name, is_active, activation_date, expiration_date, invoice_reference)
            VALUES (?, 'regold', 1, ?, ?, ?)
            ON DUPLICATE KEY UPDATE 
                is_active = 1,
                activation_date = VALUES(activation_date),
                expiration_date = VALUES(expiration_date),
                invoice_reference = VALUES(invoice_reference)
        ");
        $stmt->execute([
            $agency_id,
            parseDate($regold_activation),
            parseDate(getColumnValue($row, $headers, 'regold_expiration')),
            getColumnValue($row, $headers, 'regold_invoice')
        ]);
        $imported++;
    }
    
    // JAMES EDITION
    $james_status = getColumnValue($row, $headers, 'james_status');
    if (!empty($james_status) && $james_status !== 'Empty') {
        $stmt = $pdo->prepare("
            INSERT INTO agency_services (agency_id, service_name, is_active, expiration_date)
            VALUES (?, 'james_edition', ?, ?)
            ON DUPLICATE KEY UPDATE 
                is_active = VALUES(is_active),
                expiration_date = VALUES(expiration_date)
        ");
        $stmt->execute([
            $agency_id,
            isActive($james_status),
            parseDate(getColumnValue($row, $headers, 'james_expiration'))
        ]);
        $imported++;
    }
    
    // DOCUDROP
    $docudrop_activation = getColumnValue($row, $headers, 'docudrop_activation');
    if (!empty($docudrop_activation) && $docudrop_activation !== 'Empty') {
        $stmt = $pdo->prepare("
            INSERT INTO agency_services (agency_id, service_name, is_active, activation_date, expiration_date)
            VALUES (?, 'docudrop', 1, ?, ?)
            ON DUPLICATE KEY UPDATE 
                is_active = 1,
                activation_date = VALUES(activation_date),
                expiration_date = VALUES(expiration_date)
        ");
        $stmt->execute([
            $agency_id,
            parseDate($docudrop_activation),
            parseDate(getColumnValue($row, $headers, 'docudrop_expiration'))
        ]);
        $imported++;
    }
    
    // UNIQUE
    $unique_status = getColumnValue($row, $headers, 'unique_status');
    if (!empty($unique_status) && $unique_status !== 'Empty') {
        $stmt = $pdo->prepare("
            INSERT INTO agency_services (agency_id, service_name, is_active)
            VALUES (?, 'unique', ?)
            ON DUPLICATE KEY UPDATE is_active = VALUES(is_active)
        ");
        $stmt->execute([$agency_id, isActive($unique_status)]);
        $imported++;
    }
}

fclose($handle);

echo "\n✅ IMPORT COMPLETATO!\n";
echo "   Servizi importati: $imported\n";
echo "   Errori: $errors\n";

File: import_services.py
#!/usr/bin/env python3
"""
Import servizi agenzie da CSV a database
Mappa i campi servizi dal CSV Notion alla tabella agency_services
"""

import pandas as pd
import mysql.connector
from datetime import datetime

# Connessione database
db = mysql.connector.connect(
    host="localhost",
    user="crm_user",
    password="CRM_cb2025!Secure",
    database="crm_coldwell_banker"
)

cursor = db.cursor()

# Leggi CSV
csv_file = input("Inserisci il path del CSV agenzie: ")
df = pd.read_csv(csv_file)

print(f"Trovate {len(df)} agenzie nel CSV")
print("\nPrime 5 colonne:", list(df.columns[:5]))
print("\nCerco colonne servizi...")

# Trova colonne servizi (adatta questi nomi ai nomi REALI nel CSV)
service_columns = {
    'cb_suite': ['CB Suite', 'CB Suite dal', 'CB Suite al', 'Obbligo rinnovo', 'Fattura CB Suite'],
    'canva': ['CANVA'],
    'regold': ['ATTIVAZIONE REGOLD', 'SCADENZA REGOLD', 'Fattura Regold'],
    'james_edition': ['JamesEdition', 'SCADENZA JAMESEDITION'],
    'docudrop': ['ATTIVAZIONE DOCUDROP', 'SCADENZA DOCUDROP'],
    'unique': ['Attivazione Unique']
}

# Mostra colonne trovate
print("\nColonne nel CSV:")
for col in df.columns:
    if any(keyword.lower() in col.lower() for keyword in ['suite', 'canva', 'regold', 'james', 'docudrop', 'unique']):
        print(f"  - {col}")

print("\n" + "="*50)
confirm = input("Vuoi procedere con l'import? (si/no): ")
if confirm.lower() != 'si':
    print("Import annullato")
    exit()

# Funzione helper per convertire date
def parse_date(date_str):
    if pd.isna(date_str) or date_str == '' or date_str == 'Empty':
        return None
    try:
        return pd.to_datetime(date_str).strftime('%Y-%m-%d')
    except:
        return None

# Funzione helper per check attivo
def is_service_active(value):
    if pd.isna(value) or value == '' or value == 'Empty':
        return 0
    if str(value).lower() in ['attivo', 'active', 'si', 'yes', '1', 'true']:
        return 1
    return 0

# Import servizi
imported = 0
errors = 0

for idx, row in df.iterrows():
    agency_code = row.get('Codice', row.get('code', ''))
    
    if pd.isna(agency_code) or agency_code == '':
        continue
    
    # Trova agency_id
    cursor.execute("SELECT id FROM agencies WHERE code = %s", (agency_code,))
    result = cursor.fetchone()
    
    if not result:
        print(f"⚠️  Agenzia {agency_code} non trovata nel database")
        errors += 1
        continue
    
    agency_id = result[0]
    
    # CB Suite
    cb_suite_status = row.get('CB Suite (EuroMLS)', row.get('CB Suite', ''))
    if not pd.isna(cb_suite_status) and cb_suite_status != '' and cb_suite_status != 'Empty':
        cursor.execute("""
            INSERT INTO agency_services (agency_id, service_name, is_active, activation_date, expiration_date, renewal_required, invoice_reference, notes)
            VALUES (%s, 'cb_suite', %s, %s, %s, %s, %s, %s)
            ON DUPLICATE KEY UPDATE 
                is_active = VALUES(is_active),
                activation_date = VALUES(activation_date),
                expiration_date = VALUES(expiration_date),
                renewal_required = VALUES(renewal_required),
                invoice_reference = VALUES(invoice_reference),
                notes = VALUES(notes)
        """, (
            agency_id,
            is_service_active(cb_suite_status),
            parse_date(row.get('CB Suite dal', '')),
            parse_date(row.get('CB Suite al', '')),
            row.get('Obbligo rinnovo?', ''),
            row.get('Fattura CB Suite', ''),
            row.get('NOTE', '')
        ))
        imported += 1
    
    # Canva
    canva_status = row.get('CANVA', '')
    if not pd.isna(canva_status) and canva_status != '' and canva_status != 'Empty':
        cursor.execute("""
            INSERT INTO agency_services (agency_id, service_name, is_active)
            VALUES (%s, 'canva', %s)
            ON DUPLICATE KEY UPDATE is_active = VALUES(is_active)
        """, (agency_id, is_service_active(canva_status)))
        imported += 1
    
    # Regold
    regold_activation = row.get('ATTIVAZIONE REGOLD', '')
    if not pd.isna(regold_activation) and regold_activation != '' and regold_activation != 'Empty':
        cursor.execute("""
            INSERT INTO agency_services (agency_id, service_name, is_active, activation_date, expiration_date, invoice_reference)
            VALUES (%s, 'regold', 1, %s, %s, %s)
            ON DUPLICATE KEY UPDATE 
                is_active = 1,
                activation_date = VALUES(activation_date),
                expiration_date = VALUES(expiration_date),
                invoice_reference = VALUES(invoice_reference)
        """, (
            agency_id,
            parse_date(regold_activation),
            parse_date(row.get('SCADENZA REGOLD', '')),
            row.get('Fattura Regold', '')
        ))
        imported += 1
    
    # James Edition
    james_status = row.get('JamesEdition', '')
    if not pd.isna(james_status) and james_status != '' and james_status != 'Empty':
        cursor.execute("""
            INSERT INTO agency_services (agency_id, service_name, is_active, expiration_date)
            VALUES (%s, 'james_edition', %s, %s)
            ON DUPLICATE KEY UPDATE 
                is_active = VALUES(is_active),
                expiration_date = VALUES(expiration_date)
        """, (
            agency_id,
            is_service_active(james_status),
            parse_date(row.get('SCADENZA JAMESEDITION', ''))
        ))
        imported += 1
    
    # Docudrop
    docudrop_activation = row.get('ATTIVAZIONE DOCUDROP', '')
    if not pd.isna(docudrop_activation) and docudrop_activation != '' and docudrop_activation != 'Empty':
        cursor.execute("""
            INSERT INTO agency_services (agency_id, service_name, is_active, activation_date, expiration_date)
            VALUES (%s, 'docudrop', 1, %s, %s)
            ON DUPLICATE KEY UPDATE 
                is_active = 1,
                activation_date = VALUES(activation_date),
                expiration_date = VALUES(expiration_date)
        """, (
            agency_id,
            parse_date(docudrop_activation),
            parse_date(row.get('SCADENZA DOCUDROP', ''))
        ))
        imported += 1
    
    # Unique
    unique_status = row.get('Attivazione Unique', '')
    if not pd.isna(unique_status) and unique_status != '' and unique_status != 'Empty':
        cursor.execute("""
            INSERT INTO agency_services (agency_id, service_name, is_active)
            VALUES (%s, 'unique', %s)
            ON DUPLICATE KEY UPDATE is_active = VALUES(is_active)
        """, (agency_id, is_service_active(unique_status)))
        imported += 1

db.commit()

print(f"\n✅ Import completato!")
print(f"   Servizi importati: {imported}")
print(f"   Errori: {errors}")

cursor.close()
db.close()
